File: go-client-sdk/pkg/models/operations/postsubscriptionssubscriptionidupdatefixedfeequantity.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/v2/pkg/models/shared"
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/v2/pkg/types"
)

type PostSubscriptionsSubscriptionIDUpdateFixedFeeQuantityPathParams struct {
	SubscriptionID string `pathParam:"style=simple,explode=false,name=subscription_id"`
}

type PostSubscriptionsSubscriptionIDUpdateFixedFeeQuantityRequestBody struct {
	EffectiveDate *types.Date `json:"effective_date,omitempty"`
	PriceID       string      `json:"price_id"`
	Quantity      float64     `json:"quantity"`
}

type PostSubscriptionsSubscriptionIDUpdateFixedFeeQuantityRequest struct {
	PathParams PostSubscriptionsSubscriptionIDUpdateFixedFeeQuantityPathParams
	Request    *PostSubscriptionsSubscriptionIDUpdateFixedFeeQuantityRequestBody `request:"mediaType=application/json"`
}

type PostSubscriptionsSubscriptionIDUpdateFixedFeeQuantityResponse struct {
	ContentType  string
	StatusCode   int
	Subscription *shared.Subscription
}

File: typescript-client-sdk/src/sdk/models/operations/postsubscriptionssubscriptionidupdatefixedfeequantity.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Expose, Transform, Type } from "class-transformer";


export class PostSubscriptionsSubscriptionIdUpdateFixedFeeQuantityPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=subscription_id" })
  subscriptionId: string;
}

export class PostSubscriptionsSubscriptionIdUpdateFixedFeeQuantityRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "effective_date" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  effectiveDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "price_id" })
  priceId: string;

  @SpeakeasyMetadata()
  @Expose({ name: "quantity" })
  quantity: number;
}

export class PostSubscriptionsSubscriptionIdUpdateFixedFeeQuantityRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PostSubscriptionsSubscriptionIdUpdateFixedFeeQuantityPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostSubscriptionsSubscriptionIdUpdateFixedFeeQuantityRequestBody;
}

export class PostSubscriptionsSubscriptionIdUpdateFixedFeeQuantityResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  subscription?: shared.Subscription;
}